tapi-model/src/main/java/org/gradle/declarative/lsp/build/action/GetDeclarativeResourcesModel.java

```diff
@@ -103,7 +104,13 @@ public File getRootDir() {
         @Override
         public File getSettingsFile() {
             // TODO: this is an assumption about the location of the settings file – get it from Gradle instead.
-            return new File(getRootDir(), "settings.gradle.dcl");
+            List<File> candidateFiles = Stream.of("settings.gradle.dcl", "settings.gradle.kts")
+                    .map(it -> new File(getRootDir(), "settings.gradle.dcl"))
+                    .collect(Collectors.toList());
+            return candidateFiles.stream()
+                    .filter(File::exists)
+                    .findFirst()
+                    .orElse(candidateFiles.get(0));
```

Comment:
:x: Why collect it meanwhile you are streaming it again? I think this works, and it's much more readable.
```suggestion
            List<File> candidateFiles = Stream.of("settings.gradle.dcl", "settings.gradle.kts")
                    .map(it -> new File(getRootDir(), "settings.gradle.dcl"))
                    .filter(File::exists)
                    .findFirst()
                    .orElse(candidateFiles.get(0));
```

Reply:
`.orElse(candidateFiles.get(0))` is a self-reference in the initializer expression of `candidateFiles`, or am I not getting it right?

I wanted to avoid duplicating the call to `new File(...)` and avoid duplicating the name. To avoid both, I had to introduce a named variable to reference later, and that cannot be a stream, as you cannot iterate a stream twice. I'll see how to refactor it to make it more readable.

Reply:
Do you think this would be a better option?

```java
            List<String> candidateFileNames = Arrays.asList("settings.gradle.dcl", "settings.gradle.kts");
            Function<String, File> asFileInRootDirectory = it -> new File(getRootDir(), it);
            
            return  candidateFileNames.stream()
                    .map(asFileInRootDirectory)
                    .filter(File::exists)
                    .findFirst()
                    .orElse(asFileInRootDirectory.apply(candidateFileNames.get(0)));
```

Reply:
☝️ I think this is more readable